src/consensus/merkle.cpp

```diff
@@ -66,20 +66,20 @@ uint256 ComputeMerkleRoot(std::vector<uint256> hashes, bool* mutated) {
 uint256 BlockMerkleRoot(const CBlock& block, bool* mutated)
 {
     std::vector<uint256> leaves;
-    leaves.resize(block.vtx.size());
-    for (size_t s = 0; s < block.vtx.size(); s++) {
-        leaves[s] = block.vtx[s]->GetHash();
+    leaves.reserve((block.vtx.size() + 1) & ~1ULL); // capacity rounded up to even
```

Comment:
This assumes `size_t` is `unsigned long long` I think? Can we avoid that easily?

Reply:
we could either do `size_t{1}` or `leaves.reserve(block.vtx.size() + (block.vtx.size() & 1));`, I think, if you think this could be problematic on 32 bit systems.

Reply:
Thanks for checking, I think it's the best solution regarding C++ simplicity and CPU support.